recipes/automake/all/test_package/conanfile.py

```diff
@@ -1,89 +1,57 @@
-
-    def _build_scripts(self):
-        """Test compile script of automake"""
-        compile_script = self._user_info["automake"].compile
-        ar_script = self._user_info["automake"].ar_lib
-        assert os.path.isfile(ar_script)
-        assert os.path.isfile(compile_script)
+    def win_bash(self):
+        return self._settings_build.os == "Windows"
 
-        if self._system_cc:
-            with tools.vcvars(self) if is_msvc(self) else tools.no_op():
-                self.run("{} {} test_package_1.c -o script_test".format(tools.unix_path(compile_script), self._system_cc), win_bash=tools.os_info.is_windows)
+    def layout(self):
+        basic_layout(self, src_folder=".")
+        self.folders.build = "build"
+        self.folders.generators = "build/conan"
 
-    def _build_autotools(self):
-        """Test autoreconf + configure + make"""
-        with tools.environment_append({"AUTOMAKE_CONAN_INCLUDES": [tools.unix_path(self.source_folder)]}):
-            self.run("{} -fiv".format(os.environ["AUTORECONF"]), win_bash=tools.os_info.is_windows)
-        self.run("{} --help".format(os.path.join(self.build_folder, "configure").replace("\\", "/")), win_bash=tools.os_info.is_windows)
-        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        with self._build_context():
-            autotools.configure()
-            autotools.make()
+    def build_requirements(self):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            self.tool_requires("msys2/cci.latest")  # The conf `tools.microsoft.bash:path` and `tools.microsoft.bash:subsystem` aren't injected for test_package
+        self.tool_requires(self.tested_reference_str)
 
     def build(self):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            return  # autoconf needs a bash if there isn't a bash no need to build
+
         for src in self.exports_sources:
-            shutil.copy(os.path.join(self.source_folder, src), self.build_folder)
+            shutil.copy(self.source_path.joinpath(src), self.build_folder)
 
-        self._build_scripts()
-        self._build_autotools()
+        autotools = Autotools(self)
+        self.run("autoreconf -fiv", cwd=self.build_path, env="conanbuild")  # Workaround for since the method `autoreconf()` will always run from source
+        autotools.configure(build_script_folder=self.build_path)
+        autotools.make()
 
     def test(self):
-        if self._system_cc:
-            if not tools.cross_building(self):
-                self.run(os.path.join(".", "script_test"), run_environment=True)
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
+            return  # autoconf needs a bash if there isn't a bash no need to build
```

Comment:
I know this case is a configuration issue and not an issue with the recipe but I still think it would be good to output something here. It might save someone some searching in the future. 

```suggestion
            self.output.warn("You must have bash configured to run this test")
            return  # autoconf needs a bash if there isn't a bash no need to build
```